### PR TITLE
LSP proc re-enter due to reconnection

### PR DESCRIPTION
**Pull request by PavelSosin**

### CODE REVIEW

File: src/main/java/com/sap/lsp/cf/ws/LSPProcessManager.java [hunk 1]
```diff
@@ -219,14 +214,24 @@ public void run() {
...
 			this.remoteEndpoint = remoteEndpoint; 
 			this.projPathElem = "/" + String.join("/", Arrays.copyOfRange(wsKeyElem,1,wsKeyElem.length));
+			this.ownerSessionId = ownerSessionId;
+		}
+
+		protected String getOwnerSessionId() {
+			return ownerSessionId;
+		}
+
+		protected void setOwnerSessionId(String ownerSessionId) {
```
Comment (amiramw) [inline src/main/java/com/sap/lsp/cf/ws/LSPProcessManager.java]: never used?

File: src/main/java/com/sap/lsp/cf/ws/LSPProcessManager.java [hunk 2]
```diff
@@ -219,14 +214,24 @@ public void run() {
 		private Process process;
 		private ProcessBuilder pb;
 		private RemoteEndpoint.Basic remoteEndpoint = null;
-		private String projPathElem = ""; 
+		private String projPathElem = "";
+		private String ownerSessionId;
```
Comment (amiramw) [inline src/main/java/com/sap/lsp/cf/ws/LSPProcessManager.java]: final?

File: src/main/java/com/sap/lsp/cf/ws/LSPProcessManager.java [hunk 3]
```diff
@@ -445,6 +452,23 @@ synchronized LSPProcess createProcess(String wsKey, String lang, RemoteEndpoint.
...
+	synchronized void cleanProcess(String ws, String lang, String sessionOwnerId) {
+		String procKey = LSPProcessManager.processKey(ws, lang);
+		LSPProcess lspProc =  lspProcesses.get(procKey);
+		if ( lspProc != null && lspProc.getOwnerSessionId().equals(sessionOwnerId)) {
+			lspProc = lspProcesses.remove(procKey);
+			if (lspProc != null ) lspProc.cleanup();
+		}
+	}
+	
+	void disconnect(String procKey, String sessionOwnerId) {
```
Comment (amiramw) [inline src/main/java/com/sap/lsp/cf/ws/LSPProcessManager.java]: call it ownerSessionId

File: src/main/java/com/sap/lsp/cf/ws/LanguageServerWSEndPoint.java [hunk 4]
```diff
@@ -123,7 +123,7 @@ public void onOpen(@PathParam("ws") String ws, @PathParam("lang") String lang, S
 
        try {
             try {
-                LSPProcess process = procManager.createProcess(ws, lang, remoteEndpointBasic);
+                LSPProcess process = procManager.createProcess(ws, lang, remoteEndpointBasic, session.getId());
```
Comment (amiramw) [inline src/main/java/com/sap/lsp/cf/ws/LanguageServerWSEndPoint.java]: 1. does the session change on each handshake?
2. why do you need to check for the session? what was the problem?

File: src/main/java/com/sap/lsp/cf/ws/LSPProcessManager.java (hunk 2 above)
Comment (PavelSosin) [reply src/main/java/com/sap/lsp/cf/ws/LSPProcessManager.java]: Could you review it again?

**Comment by PavelSosin** [comment]
disconnection solution

**Pull request merged**